weaveworks/weave-gitops
# Add notifications if a new profile version is available

Closes #1143

# Testing

- Have your own Slack space and create a webhook URL outlined here: https://api.slack.com/messaging/webhooks
- Create a secret with the webhook URL
```
kubectl -n wego-system create secret generic slack-url \
--from-literal=address=https://hooks.slack.com/services/<token>
```
- Setup Slack notification for flux using the following notification provider:
```yaml
apiVersion: notification.toolkit.fluxcd.io/v1beta1
kind: Provider
metadata:
  name: slack
  namespace: wego-system
spec:
  type: slack
  channel: general
  secretRef:
    name: slack-url
```
- Port forward all the things which is the `notification-controller` and the `source-controller`
- Start UI
```
./bin/gitops ui run --helm-repo-namespace wego-system --helm-repo-name podinfo --notification-controller-address http://localhost:8081
```
- Install main components
```
./gitops install --config-repo=ssh://git@github.com/Skarlso/gitops-training.git
```
- Create a profile app
```
./bin/gitops add app --name podinfo --url https://weaveworks.github.io/profiles-examples --config-repo ssh://git@github.com/Skarlso/gitops-training.git --chart podinfo  --deployment-type helm --auto-merge
```
- Update the profile yaml file to include a new version and update the code in your gitops repo
- Observe incoming messages to your slack channel

<img width="605" alt="Screenshot 2022-01-19 at 17 23 31" src="https://user-images.githubusercontent.com/182850/150172252-f631a3d1-b1cf-439b-b4d9-6e6d4e2b6bd1.png">

Interesting.. where is my message?


_Update_:

Works!

<img width="513" alt="Screenshot 2022-01-20 at 13 45 51" src="https://user-images.githubusercontent.com/182850/150353495-1c5ba650-8901-4c4a-b7f4-eb5d94cdcce7.png">
